src/apps/loan-list/materials/material-decorator.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from "react";
+import {
+  GetMaterialManifestationQuery,
+  useGetMaterialManifestationQuery
+} from "../../../core/dbc-gateway/generated/graphql";
+import SelectableMaterial from "./selectable-material";
+import StackableMaterial from "./stackable-material";
+import { useText } from "../../../core/utils/text";
+
+interface MaterialDecoratorProps {
+  materialType: string;
+  faust: string;
+  dueDate: string;
+  loanType?: string;
+  renewableStatus?: string[];
+  amountOfMaterialsWithDueDate?: number;
+  selectDueDate?: () => void;
+  loanDate?: string;
+}
+
+export interface WorkManifestationType {
+  __typename?: "WorkManifestation" | undefined;
+  title?: string | null | undefined;
+  description: string;
+  fullTitle: string;
+  datePublished: unknown;
+  materialType: string;
+  creators: { __typename?: "Creator" | undefined; name: string }[];
+}
+
+const MaterialDecorator: React.FC<MaterialDecoratorProps> = ({
+  materialType,
+  faust,
+  dueDate,
+  loanType,
+  renewableStatus,
+  amountOfMaterialsWithDueDate,
+  selectDueDate,
+  loanDate
+}) => {
+  const t = useText();
+  const [material, setMaterial] = useState<GetMaterialManifestationQuery>();
+
+  // Create a string of authors with commas and a conjunction
+  const getAuthorName = (
```

Comment:
Is this function doing the same as [this](https://github.com/reload/dpl-react/blob/feature/dpl-react-disclosure-DDFSOEG-168/src/components/search-result-list/helpers.ts#L47).
If yes we should consolidate them.

Reply:
It doesn't "et al" the string.

Reply:
Then maybe we could expand the function like:
```ts
export const creatorsToString = (creators: string[], t: UseTextFunction, useEtAlSuffix: boolean = false) => {